jonathaneeckhout/jdungeon
# Add respawn logic and display heals

New:
- #20 
- Add a player's respawn location
- When dead, the player will respawn to the closest respawn location

Fix:
- #44 
- No visual update when consuming a healthpotion

How to test:
Test 1:
- Let an enemy (treetrunkguy) kill you
- The Die animation should play
- You died panel should pop up with a counter that decreases from 10
- Enemies should not hit you anymore
- Wait for the timer to reach 0
- While waiting you should not be able to do anything
- when the timer reached 0 you should respawn at a respawn spot (check world) and the you died panel should disappear.  

Test 2:
- Die from an enemy
- During the respawn time close the client or disconnect
- Log back in, you should have been respawned with full hp

Test 3:
- Use a healthpotion while your hp is lower, it should display +25 and your hp should be increased.

Test 4:
- When you login, you can hold enter and your player will only be spawned once

Disclaimers:
If a player dies, he/she can disconnect and connect again to get a quicker respawn. For me it's not really an issue as most of the times it takes longer than 10sec to connect again.
